lib/cocoapods/installer/analyzer.rb

```diff
@@ -457,8 +468,12 @@ def pod_target_test_only?(pod_target, pod_targets)
         if @test_pod_target_analyzer_cache.key?(key)
           return @test_pod_target_analyzer_cache[key]
         end
-        source = pod_targets.any? { |pt| pt.dependent_targets.map(&:name).include?(name) }
-        test = pod_targets.any? { |pt| pt.test_dependent_targets.map(&:name).include?(name) }
+        source = pod_targets.any? do |pt|
+          pt.dependent_targets.map(&:name).include?(name)
+        end
+        test = pod_targets.any? do |pt|
+          pt.test_dependent_targets.reject { |dpt| dpt.name == pt.name }.map(&:name).include?(name)
```

Comment:
We intentionally ignore test dependencies to ourselves. This was the bug. A test spec technically always depends on the parent pod target so when we added a dependency to an aggregate target that has 0 other dependencies it was mistakenly considered as test only.

Test case below to cover for this.